app/src/main/java/com/mpieterse/stride/ui/layout/central/viewmodels/HabitViewerViewModel.kt

```
@@ -121,6 +151,82 @@ class HabitViewerViewModel @Inject constructor(
+            )
+            habits.upsertLocal(dto)
+            lastHabit = dto
+            lastImageUrl = uploadedUrl
+            val updatedBitmap = when {
+                !updated.imageBase64.isNullOrBlank() -> base64ToBitmap(updated.imageBase64)
+                !uploadedUrl.isNullOrBlank() -> fetchBitmap(uploadedUrl)
+                else -> null
+            } ?: _state.value.habitImage
+            _state.update { it.copy(habitImage = updatedBitmap) }
+            eventBus.emit(AppEventBus.AppEvent.HabitUpdated)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)